### PR TITLE
feat: treat unhandled exceptions in handlers as 500 error responses

### PR DESCRIPTION
**Pull request by kettanaito**
- https://github.com/mswjs/interceptors/releases/tag/v0.28.1
- Also handles exceptions in interceptors in Node.js as 500 responses (https://github.com/mswjs/interceptors/releases/tag/v0.28.0). 
- Related to #2134, https://github.com/mswjs/interceptors/pull/552

## Todo

- [x] Decide how to release this. It is technically a breaking change. Requests will not error where they used to error before. Tests may fail, angry comments may rise.

### CODE REVIEW

**Comment by kettanaito** [comment]
Need to adjust some tests to reflect the Interceptors change.

**Comment by kettanaito** [comment]
The problem was that we throw an error in `onUnhandledRequest` and we intend that error to be forwarded to the process but since it happens within the request listener, it gets treated as an unhandled error and translated to the 500 error response. This hurts observability as it masks the error as an error response. 

I've merged https://github.com/mswjs/interceptors/pull/566 to allow us to opt-out from that default behavior and handle our internal errors differently (e.g. by forwarding them to the process).

File: src/core/utils/internal/devUtils.ts [hunk 1]
```diff
@@ -29,3 +29,7 @@ export const devUtils = {
   warn,
   error,
 }
+
+export class InternalError extends Error {
+  //
```
Comment (kettanaito) [inline src/core/utils/internal/devUtils.ts]: Okay now, this is embarrassing.

File: test/node/msw-api/setup-server/scenarios/on-unhandled-request/error.node.test.ts [hunk 2]
```diff
@@ -67,12 +68,7 @@ test('errors on unhandled request when using the "error" value', async () => {
...
 
-  expect(requestError.message).toBe('Failed to fetch')
-  /**
-   * @note Undici wraps fetch rejections in a generic "Failed to fetch" error,
-   * forwarding the actual rejection in the "error.cause" property.
-   */
-  expect(requestError.cause).toEqual(
+  expect(requestError).toEqual(
     new Error(
       '[MSW] Cannot bypass a request when using the "error" strategy for the "onUnhandledRequest" option.',
```
Comment (kettanaito) [inline test/node/msw-api/setup-server/scenarios/on-unhandled-request/error.node.test.ts:73]: This is a correct "breaking" change. Errors originating from MSW must not be treated as request errors.

**Comment by kettanaito** [comment]
## Released: v2.3.0 🎉

This has been released in v2.3.0!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v2.3.0)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/2.3.0)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Comment by jrnail23** [comment]
While I don't love the breaking change with a minor rev, this is a good change!
It actually allowed me to undo a bunch of special casing I introduced into my tests to handle the previously unhandled errors as I'm porting my http stubbing library from `nock` to `msw`.

**Pull request merged**